webapp/channels/src/components/widgets/inputs/input/input.tsx

```diff
@@ -43,6 +43,7 @@ export interface InputProps extends Omit<React.InputHTMLAttributes<HTMLInputElem
     clearable?: boolean;
     clearableTooltipText?: string;
     onClear?: () => void;
+    validate?: () => CustomMessageInputType | undefined;
```

Comment:
Perhaps `validate` could return `ReactNode | undefined` and so that the calling code doesn't need to specify that it's an error. I could see cases where we might want to do that, but I think it's worth it to keep the API simpler. It'd also probably mean that you don't need to fight with TS wanting you to add `as const` to `type: 'error'` in as many places

Reply:
I feel like we'd be breaking the conventions set out by the `Input` component, in which we expect that messages that are attached to the bottom of it are always some kind of error/warning/info. I wouldn't want it to be used to return some arbitrary element, as it is not the intention of the validation logic here. I think we should stick with just returning the custom message type to keep things simple.

Reply:
I missed that there's other places where we use `CustomMessageInputType` since I thought it was some internal type to this component due to the name. That makes sense then